cumulusci/tasks/github/merge.py

```diff
@@ -15,57 +15,187 @@ class MergeBranch(BaseGithubTask):
+                'children': [],
+            }
+            for child in parents.get(branch.name,[]):
+                branch_item['children'].append(branches_dict[child])
+            
+            branch_tree.append(branch_item)
+
+        # Process merge on all branches
+        for branch_item in branch_tree:
+            if self.options['children_only']:
+                if branch_item['children']:
+                    self.logger.info(
+                        'Performing merge from parent branch {} to children'.format(
+                            self.options['source_branch'],
+                        )
+                    )
+                else:
+                    self.logger.info(
+                        'No children found for branch {}'.format(
+                            self.options['source_branch'],
+                        )
+                    )
+                    continue
+                for child in branch_item['children']:
+                    self._merge(
+                        branch = child.name,
+                        source = self.options['source_branch'],
+                        commit = self.options['commit'],
+                        children = [],
+                    )
+            else:
+                self._merge(
+                    branch = branch_item['branch'].name,
+                    source = self.options['source_branch'],
+                    commit = self.options['commit'],
+                    children = branch_item['children'],
                 )
+                
+  
+    def _merge(self, branch, source, commit, children=None, indent=None): 
+        if not indent:
+            indent = ''
+        if not children:
+            children = []
+        branch_type = 'branch'
+        if children:
+            branch_type = 'parent branch'
+        if self.options['children_only']:
+            branch_type = 'child branch'
 
-                self.logger.info('Merge conflict on branch {}: created pull request #{}'.format(branch.name, pull.number))
+        compare = self.repo.compare_commits(branch, commit)
+        if not compare or not compare.files:
+            self.logger.info(
+                'Skipping {} {}: no file diffs found'.format(
+                    branch_type,
+                    branch,
+                )
+            )
+            return
+
+        try: 
+            result = self.repo.merge(branch, commit)
+            self.logger.info('{}Merged {} commits into {} {}'.format(
+                indent,
+                compare.behind_by,
+                branch_type,
+                branch,
+            ))
+            if children and not self.options['children_only']:
+                self.logger.info('  Skipping merge into the following child branches:')
+                for child in children:
+                    self.logger.info('    {}'.format(child.name))
+
+        except GitHubError as e:
+            if e.code != 409:
+                raise
```

Comment:
is there a CCI wrapped exception type we should use?

Reply:
Probably not.  github3.py is pretty good about throwing specific exceptions so the exception should provide enough info to figure out what's going on.